t/helper/test-quote.c

```diff
@@ -0,0 +1,38 @@
+#include "test-tool.h"
+#include "quote.h"
+#include "strbuf.h"
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char *nullString = 0;
+	char empty[5] = {'\0'};
+	
+	if (!argv[1]) {
+		die("missing input string");
+	}
+	
+	if (!strcmp(argv[1], "nullString")) {
+		sq_quote_buf_pretty(&buf_payload, nullString);
+
+		// Cushion the result with '.'s 
+		// to make the test script more readable
+		printf(".%s.\n", buf_payload.buf);
+		return 0;
+	}
+	
+	if (!strcmp(argv[1], "")) {
+		sq_quote_buf_pretty(&buf_payload, empty);
+		
+		// Cushion the result with '.'s 
+		// to make the test script more readable
+		printf(".%s.\n", buf_payload.buf);
+		return 0;
+	}
+	
+	sq_quote_buf_pretty(&buf_payload, argv[1]);
+	printf("%s\n", buf_payload.buf);
+	
```

Comment:
just for completeness, there should be a strbuf_release(&buf_payload) before all of the returns to avoid leaking memory.  it doesn't matter so much for tests.